### PR TITLE
WD-1958 - Applications search

### PR DESCRIPTION
**Pull request by goulinkh**
## Done

- Implement applications search
- Selectable search result
- Run an action on a list of applications

## TODO

- This feature is only supported on local apps at the moment, as I'm not sure how remote apps work

## QA

- Run the project locally
- Go to the application list 
- Perform a search
- Make sure that the selection is working properly
- Click on run action, once you have selected some applications
- Make sure that the charm panel is shown properly
- Select  a charm to run actions on
- Continue the this scenario
- Make sure that the action is executed on the appropriate applications/units

## Screenshots

image

### CODE REVIEW

**Review by huwshimi — APPROVED** [review]
This is looking great! One small improvement might be to have a way to go back to the full list of applications without having to clear the search box and then doing a blank search.

I also noticed that the search box appears when viewing units or machines, but when you do a search nothing happens:

Screenshot 2023-02-16 at 10 15 49 am

Lastly the icons appear to be rerendering every time you change the selection of applications:

Screen Recording 2023-02-16 at 10 19 57 am

**Comment by goulinkh** [comment]
@huwshimi I've fixed the issues that you mentioned, could take a look at this again?

**Comment by huwshimi** [comment]
@goulinkh sorry I made this conflict really badly!

File: src/pages/EntityDetails/Model/ApplicationsTab/useTableSelect.tsx [hunk 1]
```diff
@@ -1,66 +1,71 @@
...
     } else {
-      setSelectedApplications([...selectedApplications, application]);
+      apps = [...apps, application];
     }
+    if (apps.length === applications.length) {
+      setSelectAll(true);
+    } else if (selectAll) {
+      setSelectAll(false);
+    }
+    console.log("apps", apps);
```
Comment (huwshimi) [inline src/pages/EntityDetails/Model/ApplicationsTab/useTableSelect.tsx]: Oops left a console log here.

**Review by huwshimi — APPROVED** [review]

**Pull request merged**